clcache.py

```diff
@@ -448,15 +451,18 @@ def save(self):
 class AnalysisResult:
     Ok, NoSourceFile, MultipleSourceFilesSimple, \
         MultipleSourceFilesComplex, CalledForLink, \
-        CalledWithPch, ExternalDebugInfo = range(7)
+        CalledWithPch, ExternalDebugInfo = list(range(7))
 
 
-def getFileHash(filePath, additionalData=None):
+def getFileHash(filePath, additionalDataString=None):
     hasher = HASH_ALGORITHM()
     with open(filePath, 'rb') as inFile:
         hasher.update(inFile.read())
-    if additionalData is not None:
-        hasher.update(additionalData)
+    if additionalDataString is not None:
+        # Encoding of this additional data does not really matter
+        # as long as we keep it fixed, otherwise hashes change.
+        # The string should fit into ASCII, so UTF8 should not change anything
+        hasher.update(additionalDataString.encode("UTF-8"))
```

Comment:
I don't understand this part - how does encoding the data as UTF-8 improve anything as far as the hash is concerned? Can you give an example?